azext_iot/assets/user_messages.py

```diff
@@ -11,10 +11,13 @@
 
 
 def error_no_hub_or_login_on_input(entity_type="IoT Hub"):
-    return (
-        "Please provide an {0} entity name (via the '{1}' or '-n' parameter)"
-        " or {0} connection string via --login..."
-    ).format(entity_type, PARAM_MAPPING[entity_type])
+    if entity_type:
+        return (
+            "Please provide an {0} entity name (via the '{1}' or '-n' parameter)"
+            " or {0} connection string via --login..."
+        ).format(entity_type, PARAM_MAPPING[entity_type])
+    else:
+        return ("Required entity type missing, please check _validators.py for required entities.")
```

Comment:
This is a CLI tool not an sdk, it doesn't make sense to require a user to figure out how a CLI extension is installed, then inspect a python module to understand how validation works. If this condition is hit from a use case, then it should be re-tooled or refactored such that the user will have a clear understanding of what is required.

Reply:
This is an error message that only a developer should see (especially if they are working with offline commands and forget to update the validator). Would rewording it to "Required entity type missing." be enough for now?